base/opaque_closure.jl

```diff
@@ -52,35 +53,42 @@ function compute_oc_argtypes(ir, nargs, isva)
             push!(argtypes, Vararg{Any})
         end
     end
-    Tuple{argtypes...}
+    return Tuple{argtypes...}
 end
 
-function Core.OpaqueClosure(ir::IRCode, env...;
-        nargs::Int = length(ir.argtypes)-1,
-        isva::Bool = false,
-        rt = compute_ir_rettype(ir),
-        do_compile::Bool = true)
-    if (isva && nargs > length(ir.argtypes)) || (!isva && nargs != length(ir.argtypes)-1)
-        throw(ArgumentError("invalid argument count"))
-    end
+function Core.OpaqueClosure(ir::IRCode, @nospecialize env...;
+                            isva::Bool = false,
+                            do_compile::Bool = true)
+    # NOTE: we need ir.argtypes[1] == typeof(env)
     ir = Core.Compiler.copy(ir)
+    nargs = length(ir.argtypes)-1
+    sig = compute_oc_signature(ir, nargs, isva)
+    rt = compute_ir_rettype(ir)
     src = ccall(:jl_new_code_info_uninit, Ref{CodeInfo}, ())
-    src.slotflags = UInt8[]
     src.slotnames = fill(:none, nargs+1)
+    src.slotflags = UInt8[zero(UInt8) for _ in 1:length(ir.argtypes)]
```

Comment:
```suggestion
    src.slotflags = fill(zero(UInt8), length(ir.argtypes))
```

Reply:
Is there any advantage for using `fill` over the typed array complehension?

Reply:
Just cleaner, when applicable